doc/design/regularization.md

```diff
@@ -0,0 +1,103 @@
+# Regularization in PaddlePaddle
+
+## Introduction to Regularization
+A central problem in machine learning is how to design an algorithm that will perform well not just on the training data, but also on new data. Many strategies are used by machine learning practitioners to reduce the test error, possibly at the expense of increased training error. These strategies are collectively known as **regularization**. 
+
```

Comment:
Maybe we can introduce the `overfit` problem and then comes out the `regularization`.
>Many strategies are used by machine learning practitioners to reduce the test error, possibly at the expense of increased training error.

These strategies summarization of `regularization` is not very accurate to my first glance.  
from the wikipedia, 
[regularization](https://en.wikipedia.org/wiki/Regularization_(mathematics))

>In general, regularization is a technique that applies to objective functions in ill-posed problems formulated as optimization problems